lib/device_set.js

```diff
@@ -1,539 +1,1106 @@
+							this.devicesById[patch._id].handlePatch(patch)
+						}
+					})
+				} else if (msg.message.da.seca || msg.message.da.secd) {
+          this.handleSecurity((msg.message.da.seca || msg.message.da.secd), 0)
         }
-        msg.message.da.d.forEach((patch) => {
-          if (this.devicesById[patch._id]) {
-            this.devicesById[patch._id].handlePatch(patch)
-          }
-        })
-      }
-    }
-  }
-
-  class Device {
-    constructor(accessory) {
-      this.id = accessory.context.id
-      this.name = accessory.context.name
-      this.data = {}
-    }
-
-    handleSnapshot(data) {
-      if (data._id != this.id)
-        throw "This snapshot does not belong to this device"
-      this.data = data
-      this.notify()
-    }
-
-    /**
-     * Handle a PubSub patch
-     */
-    handlePatch(patch) {
-      if (patch._id != this.id)
-        throw "This patch does not belong to this device"
-
-      dataPatch(this.data, patch)
-      this.notify()
-    }
-
-    notify() {
-      throw new("notify not implemented")
-    }
-  }
-
-  extend(DeviceSet, {
-    createDeviceAccessory: (data) => {
-      let deviceClass = Devices.find((dc) => { return dc.appliesTo(data) })
-      if (!deviceClass)
-        return null // we don't know how to handle this device.
-
-      let name = data.n
-      let id = data._id
-      let serial = serialNumber(data.ser, data.ser32)
-      let category = deviceClass.inferCategory && deviceClass.inferCategory(data) || Accessory.Categories.OTHER
-
-      let accessory = new PlatformAccessory(
-        name,
-        uuid.generate("Vivint:" + id + ":" + serial),
-        category) // setting category doesn't seem to work :'(
-
-      accessory.context.name = name
-      accessory.context.id = data._id
-      accessory.context.deviceClassName = deviceClass.name
-
-      let informationService = accessory.getServiceByUUIDAndSubType(Service.AccessoryInformation)
-
-      informationService
-        .setCharacteristic(Characteristic.Manufacturer, "Vivint")
-        .setCharacteristic(Characteristic.Model, deviceClass.name)
-        .setCharacteristic(Characteristic.SerialNumber, serial);
-      
-      deviceClass.addServices(accessory)
-      return accessory
-    }
-  })
-
-  class ContactSensor extends Device {
-    constructor(accessory) {
-      super(accessory)
-      this.service = accessory.getServiceByUUIDAndSubType(Service.ContactSensor)
-
-      this.service
-        .getCharacteristic(Characteristic.ContactSensorState)
-        .on('get', (next) => next(null, this.contactSensorValue()));
-    }
-
-    contactSensorValue() {
-      if (this.data.s)
-        return Characteristic.ContactSensorState.CONTACT_NOT_DETECTED
-      else
-        return Characteristic.ContactSensorState.CONTACT_DETECTED
-    }
-
-    notify() {
-      this.service.getCharacteristic(Characteristic.ContactSensorState)
-        .updateValue(this.contactSensorValue())
-    }
-  }
-
-  extend(ContactSensor, {
```

Comment:
Can you try and match the existing code styling? (is there a JavaScript code formatter we can use?)

I don't really care how it is formatted, so much that it is consistent and that unrelated lines aren't changed.

Reply:
I had ran it through a beautifier at https://beautifier.io ... sorry I've used a couple different text editors and landed on atom at the moment. Apparently I had made some changes and forgot to run it through the beautifier again! Going to see if I can resolve the dimmer issue first. Do I need to close the PR and reopen a new one? or will it update if I upload the file again?